src/providers/CartridgesView.ts

```diff
@@ -0,0 +1,146 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export class CartridgesView implements vscode.TreeDataProvider<CartridgeItem> {
+	private _onDidChangeTreeData: vscode.EventEmitter<CartridgeItem | undefined> = new vscode.EventEmitter<CartridgeItem | undefined>();
+	readonly onDidChangeTreeData: vscode.Event<CartridgeItem | undefined> = this._onDidChangeTreeData.event;
+
+	constructor(private workspaceRoot: string) {
+	}
+
+	refresh(): void {
+		this._onDidChangeTreeData.fire();
+	}
+
+	getTreeItem(element: CartridgeItem): vscode.TreeItem {
+		return element;
+	}
+
+	getChildren(element?: CartridgeItem): Thenable<CartridgeItem[]> {
+		if (!this.workspaceRoot) {
+			vscode.window.showInformationMessage('No dependency in empty workspace');
+			return Promise.resolve([]);
+		}
+
+		return new Promise(resolve => {
+			if (element) {
+				resolve(this.getCartridgeItemFilesOrFolders(element));
+			} else {
+				if (this.pathExists(this.workspaceRoot)) {
+					resolve(this.getCartridgesInWorkspace(this.workspaceRoot));
+				} else {
+					vscode.window.showInformationMessage('No workspace!');
+					resolve([]);
+				}
+			}
+		});
+	}
+
+	private getCartridgeItemFilesOrFolders(element: CartridgeItem): CartridgeItem[] {
+		var files = this.getFiles(element.location);
+		var directories = this.getDirectories(element.location);
+
+		if (files.length > 0 || directories.length > 0) {
+			const toFileElement = (fileName: string): CartridgeItem => {
+				return new CartridgeItem(fileName, 'cartridge-item-file', path.join(element.location, fileName), vscode.TreeItemCollapsibleState.None, {
+					command: 'vscode.open',
+					title: 'Open file',
+					arguments: [vscode.Uri.parse('file://' + path.join(element.location, fileName))],
+				});
+			}
+
+			const toFolderElement = (directory: string): CartridgeItem => {
+				return new CartridgeItem(directory, 'cartridge-item-folder', path.join(element.location, directory), vscode.TreeItemCollapsibleState.Collapsed);
+			}
+
+			return directories.map(toFolderElement).concat(files.map(toFileElement));
+		}
+
+		return [new CartridgeItem('No files', 'cartridge-file', '', vscode.TreeItemCollapsibleState.None)];
+	}
+
+
+	private getCartridgesInWorkspace(workspaceRoot: string): CartridgeItem[] {
+		if (this.pathExists(workspaceRoot)) {
+			var directories = this.getDirectories(workspaceRoot)
```

Comment:
Not always cartridges root and `workspaceRoot` is the same folder. For example in my company we use next structure:

```
├── cartridges <-- contains all cartridges
├── dw.json
├── dw.json.example
├── gulpfile.js
├── node_modules
├── package.json
├── package-lock.json
├── README.md
├── ssl
└── types
```
Hence need the better way to detect that case.

PS. Also, would be great considered `modules` folder as cartridge too.